### DISCUSSION TITLE
I don't see any key data in React query Devtools after data is fetched , It only show in devtools when Route changed

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [TanStack/query#5801] [Q&A]
I just want to delete firestore data and update the UI . So I tried invalidateQueries , reFetchQuery and mutate onSuccess . It only work when data are in devtools . So I checked devtools And I noticed that data  is not in dev tools even data is already fetched . So when data arrive in devtools ? I tested . It only show data in devtools, after route is changed. 

### 1
I think Problem is solved when using useState with queryClient ()

`const [queryClient] = useState(() => new QueryClient());
`

**Comment by TkDodo** [comment]
https://tkdodo.eu/blog/react-query-fa-qs#2-the-queryclient-is-not-stable
